### ISSUE TITLE
SPIR-V: invalid codegen

### ISSUE THREAD

**Issue by kennyalive** [Microsoft/DirectXShaderCompiler#1137]
For the following shader the generated SPIRV uses value.xy for computations instead of value.zw:

```
float4 main(in float4 value : TEXCOORD0) : SV_POSITION
{
	float2 v = value.zw;
	return float4(v.x + v.y, 0, 0, 1);
}

```
Command line arguments: -O4 -spirv -Tvs_5_1

Generated SPIRV:

```
; SPIR-V
; Version: 1.0
; Generator: Google spiregg; 0
; Bound: 25
; Schema: 0
               OpCapability Shader
               OpMemoryModel Logical GLSL450
               OpEntryPoint Vertex %main "main" %gl_PerVertexOut %in_var_TEXCOORD0
               OpName %main "main"
               OpName %type_gl_PerVertex "type.gl_PerVertex"
               OpName %gl_PerVertexOut "gl_PerVertexOut"
               OpName %in_var_TEXCOORD0 "in.var.TEXCOORD0"
               OpMemberDecorate %type_gl_PerVertex 0 BuiltIn Position
               OpMemberDecorate %type_gl_PerVertex 1 BuiltIn PointSize
               OpMemberDecorate %type_gl_PerVertex 2 BuiltIn ClipDistance
               OpMemberDecorate %type_gl_PerVertex 3 BuiltIn CullDistance
               OpDecorate %type_gl_PerVertex Block
               OpDecorate %in_var_TEXCOORD0 Location 0
       %void = OpTypeVoid
          %6 = OpTypeFunction %void
      %float = OpTypeFloat 32
    %v4float = OpTypeVector %float 4
       %uint = OpTypeInt 32 0
     %uint_1 = OpConstant %uint 1
%_arr_float_uint_1 = OpTypeArray %float %uint_1
%type_gl_PerVertex = OpTypeStruct %v4float %float %_arr_float_uint_1 %_arr_float_uint_1
%_ptr_Output_type_gl_PerVertex = OpTypePointer Output %type_gl_PerVertex
%_ptr_Input_v4float = OpTypePointer Input %v4float
%_ptr_Output_v4float = OpTypePointer Output %v4float
     %uint_0 = OpConstant %uint 0
    %float_0 = OpConstant %float 0
    %float_1 = OpConstant %float 1
%gl_PerVertexOut = OpVariable %_ptr_Output_type_gl_PerVertex Output
%in_var_TEXCOORD0 = OpVariable %_ptr_Input_v4float Input
       %main = OpFunction %void None %6
         %18 = OpLabel
         %19 = OpLoad %v4float %in_var_TEXCOORD0
         %20 = OpCompositeExtract %float %19 0
         %21 = OpCompositeExtract %float %19 1
         %22 = OpFAdd %float %20 %21
         %23 = OpCompositeConstruct %v4float %22 %float_0 %float_0 %float_1
         %24 = OpAccessChain %_ptr_Output_v4float %gl_PerVertexOut %uint_0
               OpStore %24 %23
               OpReturn
               OpFunctionEnd

```

Unoptimized SPIRV (with -O0) produces correct code.

**Comment by antiagainst** [comment]
Known bug in SPIRV-Tools https://github.com/KhronosGroup/SPIRV-Tools/issues/1385 that recently got fixed by https://github.com/KhronosGroup/SPIRV-Tools/pull/1386.

I'll refresh SPIRV-Tools to pull in the fix.

**Comment by antiagainst** [comment]
Should be fixed by #1139
